Tests/Packet++Test/Tests/IcmpV6Tests.cpp

```diff
@@ -0,0 +1,287 @@
+	PTF_ASSERT_EQUAL(neighAdvNoOptLayer->getTargetIP(), pcpp::IPv6Address("fe80:ebeb:ebeb::1"));
+	PTF_ASSERT_FALSE(neighAdvNoOptLayer->hasTargetMacInfo());
+	PTF_ASSERT_EQUAL(neighAdvNoOptLayer->getTargetMac(), pcpp::MacAddress());
+	PTF_ASSERT_EQUAL(neighAdvNoOptLayer->getNdpOptionCount(), 0);
+	pcpp::NdpOption targetLinkLayerNoOptOption = neighAdvNoOptLayer->getNdpOption(pcpp::NDPNeighborOptionTypes::NDP_OPTION_TARGET_LINK_LAYER);
+	PTF_ASSERT_TRUE(targetLinkLayerNoOptOption.isNull());
+}
+
+PTF_TEST_CASE(IcmpV6CreationTest)
+{
+	timeval time;
+	gettimeofday(&time, NULL);
+
+	READ_FILE_INTO_BUFFER(1, "PacketExamples/IcmpV6_EchoRequest.dat");
+	READ_FILE_INTO_BUFFER(2, "PacketExamples/IcmpV6_EchoReply.dat");
+	READ_FILE_INTO_BUFFER(3, "PacketExamples/IcmpV6_NeighSoli.dat");
+	READ_FILE_INTO_BUFFER(4, "PacketExamples/IcmpV6_NeighAdv.dat");
+	READ_FILE_INTO_BUFFER(5, "PacketExamples/IcmpV6_NeighAdvNoOption.dat");
+
+	uint8_t data[56] = {0xbd, 0xce, 0xcb, 0x62, 0x00, 0x00, 0x00, 0x00, 0xf3, 0xa1, 0x09, 0x00, 0x00, 0x00,
+						0x00, 0x00, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x1a, 0x1b,
+						0x1c, 0x1d, 0x1e, 0x1f, 0x20, 0x21, 0x22, 0x23, 0x24, 0x25, 0x26, 0x27, 0x28, 0x29,
+						0x2a, 0x2b, 0x2c, 0x2d, 0x2e, 0x2f, 0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37};
+
+	pcpp::EthLayer ethLayer(pcpp::MacAddress("11:22:33:44:55:66"), pcpp::MacAddress("66:55:44:33:22:11"));
+	pcpp::IPv6Layer ipv6Layer(pcpp::IPv6Address(std::string("fe80::215:5dff:fea5:c4c5")), pcpp::IPv6Address(std::string("fe80::dd05:dae0:74bc:7341")));
+
+	// Echo request creation
+	pcpp::ICMPv6EchoLayer echoReqLayer(pcpp::ICMPv6EchoLayer::REQUEST, 0x0018, 20, data, 56);
+	pcpp::Packet echoRequestPacket(100);
+	PTF_ASSERT_TRUE(echoRequestPacket.addLayer(&ethLayer));
+	PTF_ASSERT_TRUE(echoRequestPacket.addLayer(&ipv6Layer));
+	PTF_ASSERT_TRUE(echoRequestPacket.addLayer(&echoReqLayer));
+	echoRequestPacket.computeCalculateFields();
+	PTF_ASSERT_EQUAL(echoRequestPacket.getRawPacket()->getRawDataLen(), bufferLength1);
+	PTF_ASSERT_BUF_COMPARE(echoRequestPacket.getRawPacket()->getRawData()+54, buffer1+54, bufferLength1-54);
+
+	// Echo reply creation
+	pcpp::EthLayer ethLayer2(ethLayer);
+	pcpp::IPv6Layer ipLayer2(ipv6Layer);
+	pcpp::ICMPv6EchoLayer echoRepLayer(pcpp::ICMPv6EchoLayer::REPLY, 0x0018, 20, data, 56);
+	pcpp::Packet echoReplyPacket(100);
+	PTF_ASSERT_TRUE(echoReplyPacket.addLayer(&ethLayer2));
+	PTF_ASSERT_TRUE(echoReplyPacket.addLayer(&ipLayer2));
+	PTF_ASSERT_TRUE(echoReplyPacket.addLayer(&echoRepLayer));
+	echoReplyPacket.computeCalculateFields();
+	PTF_ASSERT_EQUAL(echoReplyPacket.getRawPacket()->getRawDataLen(), bufferLength2);
+	PTF_ASSERT_BUF_COMPARE(echoReplyPacket.getRawPacket()->getRawData()+54, buffer2+54, bufferLength2-54);
+
+	// Neighbor solicitation with source link-layer option
+	pcpp::IPv6Layer *pIpv6SoliLayer = new pcpp::IPv6Layer(pcpp::IPv6Address("fd53:7cb8:383:4::67"), pcpp::IPv6Address("fd53:7cb8:383:2::1:117"));
```

Comment:
nit: we usually don't start a pointer variable with a `p`, I'd rename `pIpv6SoliLayer` to `ipv6SoliLayer`